src/test/java/edu/gatech/VRDR/MessagingTest.java

```diff
@@ -879,5 +881,44 @@ public void testCreateSubmissionWithRaceLiterals() {
         DeathRecordSubmissionMessage parsed = BaseMessage.parseJson(DeathRecordSubmissionMessage.class, ctx, submissionBundleStr);
         assertNotNull(parsed); // make sure we can parse the race values without crashing
     }
+	
+	public void testCreateBulkUploadPayload() {
+		// create a new death certificate "deathCertificate1" and add it to a death record
+		DeathCertificate deathCertificate1 = new DeathCertificate();
+		DeathCertificateDocument deathRecord1 = new DeathCertificateDocument();
+		CommonUtil.initResource((Resource)deathCertificate1);
+		deathRecord1.addEntry((new Bundle.BundleEntryComponent()).setResource((Resource)deathCertificate1));
+		
+		// create a new death certificate "deathCertificate2" and add it to a death record
+		DeathCertificate deathCertificate2 = new DeathCertificate();
+		DeathCertificateDocument deathRecord2 = new DeathCertificateDocument();
+		CommonUtil.initResource((Resource)deathCertificate2);
+		deathRecord2.addEntry((new Bundle.BundleEntryComponent()).setResource((Resource)deathCertificate2));
+		
+		// create a new submission message "message1" and add the death record to it
+		DeathRecordSubmissionMessage message1 = new DeathRecordSubmissionMessage();
+		message1.setDeathRecord(deathRecord1);
+		message1.setCertNo(Integer.valueOf(42));
+		message1.setStateAuxiliaryId("identifier");
+		
+		// create a new submission message "message2" and add the death record to it
+		DeathRecordSubmissionMessage message2 = new DeathRecordSubmissionMessage();
+		message2.setDeathRecord(deathRecord2);
+		message2.setCertNo(Integer.valueOf(43));
+		message2.setStateAuxiliaryId("identifier");
+		
+		// create a list and add the messages to it
+		List<BaseMessage> messages = new ArrayList<>();
+		messages.add(message1);
+		messages.add(message2);
```

Comment:
This is good but can you write a loop to add 100 messages (just using a template and increment the certNo, also add a jurisdiction id of "TT" for testing, and set a random state auxiliary id), because that's typically the bundle size we use.  Update the count checks accordingly.

Reply:
Done. Please check. Thank you.